src/store/modules/announcements.js

```diff
@@ -0,0 +1,50 @@
+import api from '../../api/api';
+
+export default {
+  namespaced: true,
+  state: {
+    announcements: [],
+  },
+  getters: {
+    getSitewideAnnouncements(state) {
+      return state.announcements.filter(
+        ann => !('event_id' in ann),
+      );
+    },
+    getEventAnnouncements(state, eventId) {
+      return state.announcements.filter(
+        ann => ('event_id' in ann) && ann.event_id === eventId,
+      );
+    },
+  },
+  mutations: {
+    loadAnnouncements(state, { announcements }) {
+      state.announcements = [];
```

Comment:
`state.announcements = announcements`?

Reply:
wait, is it supposed to push and not replace existing data?

Reply:
Yeah this behavior should change. I did this originally to avoid duplicates but there's no need to erase the existing announcements, we could just check if an announcement already exists as it's being loaded. Will change.